File: backend/src/main/java/kr/co/kalpa/pcms/domain/apnode/ApNode.java
package kr.co.kalpa.pcms.domain.apnode;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ApNode {
    private String id;
    private String nodeType;     // F:파일, D:디렉토리, L:링크
    private String parentId;
    private String name;
    private int depth;
    private boolean deleted;
    private OffsetDateTime deleteDt;
    private OffsetDateTime createDt;
    private OffsetDateTime modifyDt;
    private int childCount;
    private long totalSize;
    private String linkTargetId;
}

File: frontend/src/domain/apnode/types/apnode.ts
export type NodeType = 'F' | 'D' | 'L'

export interface ApNode {
  id: string
  nodeType: NodeType
  parentId: string | null
  name: string
  depth: number
  createDt: string
  modifyDt: string
  // D 전용
  childCount: number
  totalSize: number
  // L 전용
  linkTargetId?: string
  brokenLink?: boolean
  // 검색 결과 전용
  pathStr?: string
  // F / L(resolved) 공통
  fileUrl?: string
  thumbnailUrl?: string
  originalName?: string
  fileSize?: number
  contentType?: string
  width?: number
  height?: number
}

export interface CtxMenu {
  show: boolean
  x: number
  y: number
  node: ApNode | null
}

export interface Clipboard {
  items: { id: string; name: string }[]
  type: 'cut' | 'copy'
}
